chemprop/uncertainty/uncertainty_predictor.py

```diff
@@ -1250,6 +1251,142 @@ def get_uncal_output(self):
         return self.uncal_confidence
 
 
+class DirichletPredictor(UncertaintyPredictor):
+    """
+    Dirichlet uncertainty
+    """
+
+    @property
+    def label(self):
+        return "dirichlet_uncal_uncertainty"
+
+    def raise_argument_errors(self):
+        super().raise_argument_errors()
+        if self.loss_function != "dirichlet":
+            raise ValueError(
+                "In order to use Dirichlet evidential uncertainty, trained models must have used dirichlet loss function."
+            )
+        if self.dataset_type not in ["classification", "multiclass"]:
+            raise ValueError(
+                "Dirichlet evidential epistemic uncertainty is only compatible with classification dataset types."
```

Comment:
Would be nice to see what we currently have with something like `f"Dirichlet ... types. Current dataset is of type {self.dataset_type}"`